### PR TITLE
Implement Union analysis

### PR DESCRIPTION
**Pull request by matriv**
@mfussenegger Please review

### CODE REVIEW

File: sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java [hunk 1]
```diff
@@ -83,12 +86,54 @@ public AnalyzedRelation analyze(Node node, Analysis analysis) {
 
     @Override
     protected AnalyzedRelation visitQuery(Query node, StatementAnalysisContext context) {
+        // In case of Set Operation
+        if (!node.getOrderBy().isEmpty() || node.getLimit().isPresent() || node.getOffset().isPresent()) {
+            // In the Future this can be a TwoRelationsSetOperation
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java]: Not sure if the future-driven comment here makes sense unless you actually explain why and how so I'd rather remove it

File: sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java [hunk 2]
```diff
@@ -83,12 +86,54 @@ public AnalyzedRelation analyze(Node node, Analysis analysis) {
...
         return process(node.getQueryBody(), context);
     }
 
     @Override
     protected AnalyzedRelation visitUnion(Union node, StatementAnalysisContext context) {
-        throw new UnsupportedOperationException("UNION is not supported");
+        if (node.isDistinct()) {
+            throw new UnsupportedOperationException("UNION [DISTINCT] is not supported");
+        } else {
+            assert node.getRelations().size() == 2 : "Each Union can have only two relations";
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java]: maybe add a comment that the parser builds a tree and ensures this?

File: sql/src/main/java/io/crate/analyze/SourceRelation.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze;
+
+import io.crate.analyze.relations.AnalyzedRelation;
+import io.crate.sql.tree.QualifiedName;
+
+public class SourceRelation {
```
Comment (dobe) [inline sql/src/main/java/io/crate/analyze/SourceRelation.java]: Would probably better to call this something like NamedRelation, however it is no relation at all, hm

File: sql/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java [hunk 4]
```diff
@@ -871,12 +872,86 @@ public void testJoinWithMultiRelationOrderBy() throws Exception {
...
+    @Test
+    public void testUnion3Tables() throws Exception {
+        SelectAnalyzedStatement analysis = analyze("select id, text from users u1 " +
+                                                   "union all " +
+                                                   "select id, name from users_multi_pk " +
+                                                   "union all " +
+                                                   "select id, name from users " +
+                                                   "order by text " +
+                                                   "limit 10 offset 20");
+        assertThat(analysis.relation(), instanceOf(TwoRelationsUnion.class));
```
Comment (dobe) [inline sql/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java]: probably use isSQL for comparison to make it more readable

**Comment by mfussenegger** [comment]
lgtm

**Pull request merged**